framework/doc/content/source/controls/WebServerControl.md

```diff
@@ -0,0 +1,122 @@
+# WebServerControl
+
+The `WebServerControl` object is designed to allow an external process to control a MOOSE solve. It works by starting up a webserver on the chosen [!param](/Controls/WebServerControl/port) and exposing a REST API. As with all REST APIs the input and output are both JSON.
+
+It can then be managed via the [MooseControl](MooseControl/index.md optional=true) python utility.
+
+## API
+
+The `WebServerControl` presents REST endpoints to help control the solve.
+
+For the endpoints that follow, if the HTTP status code received is a non-successful code, there may be a response of type `application/json` of the form:
+
+```language=json
+{
+  "error": "<ERROR>"
+}
+```
+
+where `<ERROR>` is a useful error message pertaining to the error that was encountered. The response is not guaranteed to contain `application/json` data, but it will when more context is available.
+
+The REST endpoints are as follows:
+
+### `check`
+
+Used to determine if the MOOSE webserver is currently listening. This doesn't necessarily mean that it is waiting for input.
+
+Interact with this endpoint by a `GET` request to `/check`. On success, the status code will be 200.
+
+### `waiting`
+
+Used to determine of MOOSE is currently waiting within the `WebServerControl`, i.e., that `Control` is waiting for external input. The rest of the endpoints that follow require that the control is currently waiting for input.
+
+Interact with this endpoint by a `GET` request to `/waiting` without any data. If the control is currently waiting, the response will be of type `application/json` of the form:
+
+```language=json
+{
+  "waiting": True
+  "execute_on_flag": "<EXEC_ON_FLAG>"
+}
+```
+
+where `<EXEC_ON_FLAG>` is the current execution flag. If the control is not currently waiting, the response will be a response of type `application/json` of the form:
+
+```language=json
+{
+  "waiting": False
+}
+```
+
+This endpoint can be accessed via the [MooseControl](MooseControl/index.md optional=true) python utility via the following methods:
+
+- `wait()`: Waits for the control to be waiting
+- `getWaitingFlag()`: Gets the current flag that the control is waiting on, if any
+- `isWaiting()`: Whether or not the control is currently waiting
+
+### `get/postprocessor`
+
+Used to obtain the value of a postprocessor. The control must be waiting in order to access this endpoint.
```

Comment:
Not a fan of this design at all. Why not queue these requests and have the server reply as soon as the control is in `waiting` status. The docs make it sound like you have to poll the server while the simulation is running and the control is not yet waiting.